asset/swift.go

```diff
@@ -54,3 +54,25 @@ func (s *SwiftFS) RemoveAsset(shasum string) error {
 
 	return nil
 }
+
+func (s *SwiftFS) EmptyThenDeleteContainer(name string) error {
+	sc := s.Connection
+
+	// First emptying the container
+	objs, err := sc.ObjectNames(name, nil)
+	if err != nil {
+		return err
+	}
+
+	_, err = sc.BulkDelete(name, objs)
+	if err != nil {
+		for _, obj := range objs {
+			err = sc.ObjectDelete(name, obj)
+			if err != nil {
+				return err
+			}
+		}
+	}
+
+	return sc.ContainerDelete(name)
```

Comment:
In some cases where swift is overloaded, bulk delete returns ok if the object has ben deleted from one storage server only. In this case, you cannot delete container until all objects has been purged from all object servers. You should ass a retry on `ContainerDelete()` in case it fails just after the  `BulkDelete()`.
In particular, see what have been done in cozy-stack: https://github.com/cozy/cozy-stack/blob/master/model/vfs/vfsswift/swift.go#L42-L54